### DISCUSSION TITLE
兰伯特投影

### DISCUSSION THREAD

**Discussion by sc1233** [gmt-china/GMT_docs#1272] [Q&A] [answered]
请问一下 如何解决红框内切割不完整的问题吗
对于强迫症的我总感觉差强人意
代码如下
@echo off
REM Date:    2024-01-29T22:15:38
REM User:    w89
REM Purpose: Purpose of this script
REM Set a unique session name:
set GMT_SESSION_NAME=13568
gmt begin figurename png
gmt set MAP_FRAME_TYPE fancy FORMAT_GEO_MAP ddd:mm:ssF MAP_GRID_CROSS_SIZE_PRIMARY 0.15c
gmt coast -R-180/179.9/-60/90 -JL10/35/37/65/30c  -Df -N1/0.5p,red -N2/thin -Gtan -W0.1p,white -Sblue
gmt end show
![image](https://github.com/gmt-china/GMT_docs/assets/158250558/fff86cf1-151d-41f3-a762-69aed0c97d9f)

**Comment by ZMAlt** [comment, marked as answer]
我也试了，出现了同样的问题，可能是 bug，但我认为也不必修复。不同的投影是针对不同的应用的，你这里使用的 -JL 是圆锥投影，不是很适合绘制几乎全部的南北半球范围的。
另外要明白投影参数的意义，`-JLlon0/lat0/lat1/lat2`，这里的 lon0/lat0 是投影中心，通常是位于绘制范围中心的，我想你把 180 改成 179.9 是因为报错，只要把 lon0 按报错改成 0 就可以了。绘制 360 度范围，gmt 强制投影中心经度必须是 0。至于投影中心的纬度这里也不合适，应该是 -R 为纬度范围的中心或靠近中心的位置，以防止一侧纬度变形太大。一旦确定了投影中心纬度，应该在投影中心的两个等距的位置选择两条标准纬线，也就是割圆锥的地方。

```bash
gmt begin figurename pdf
gmt coast -R-180/180/-60/90 -JL0/15/0/30/30c -Gtan -W0.1p,white -Sblue -BWSen -Baf
gmt end show
```
![figurename](https://github.com/gmt-china/GMT_docs/assets/35285040/637840bf-8999-4fdd-ab36-9c518fdc9100)

如果非要不按投影的规则来，当然也可以修改参数，做成类似下面的。如果你仔细看的话，我绘制的上一个图明显比下一个图的变形控制的更好，也比你的图好。

![figurename](https://github.com/gmt-china/GMT_docs/assets/35285040/930d96de-b550-4b52-80d8-6af0a2382e4b)

**Comment by ZMAlt** [comment]
还有，这里不是很适合发布这种问题。其实最合适的是 gmt 官方的论坛，但是没有中文论坛，官方论坛也没有专门的中文板块，或者可以在中文 QQ 群询问。
